src/validation/schema.js

```diff
@@ -339,10 +582,39 @@ const validateTypeDefinition = (defs, def) =>
 const validateTypeDefinitions = defs =>
   defs.reduce((errors, def) => errors.concat(validateTypeDefinition(defs, def)), List())
 
+const validateNamingCollisionsInTypes = types => {
+  let seen = Set()
+  let errored = Set()
+  return types.reduce((errors, type) => {
+    if (seen.has(type) && !errored.has(type)) {
+      errors = errors.push(
+        immutable.fromJS({
+          loc: { start: 1, end: 1 },
+          entity: type,
+          message: `Type ${type} is defined more than once`,
```

Comment:
I have the feeling that this error message is not going to survive very long, as it will not be helpful enough.

It may be worth to maintain the notion of local and imported types here and tell users when a type is defined multiple times in the subgraph schema or when a type conflicts with another schema (and where that was imported from).